File: src/dataset/javascript/hamming_exercism-javascript-ThomasZumsteg/claude/ClaudeSonnet4_hamming_v4.js
var Hamming = function() {};

Hamming.prototype.compute = function( dnaA, dnaB ) {
  if (dnaA.length !== dnaB.length) { throw 'DNA strands must be of equal length.'}
  
  var sum = 0;
  var length = dnaA.length;
  
  for( var i = 0; i < length; i++) {
    if (dnaA[i] !== dnaB[i]) {sum++}
  }
  return sum;
}

export default Hamming;

File: src/dataset/c/hamming_Exercism_HeitorMP/claude/ClaudeSonnet4_hamming_v4.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs)
{
    const char *p1 = lhs;
    const char *p2 = rhs;
    
    while (*p1 && *p2 && *p1 == *p2) {
        p1++;
        p2++;
    }
    
    if ((*p1 == '\0') != (*p2 == '\0'))
        return -1;
    
    int hamming = 0;
    while (*p1) {
        if (*p1 != *p2)
            hamming++;
        p1++;
        p2++;
    }
    
    return hamming;
}